bookshelf/src/main/webapp/base.jsp

```diff
@@ -51,7 +53,7 @@ Copyright 2015 Google Inc. All Rights Reserved.
           <a href="/login">Login</a>
```

Comment:
I changed the `c:otherwise` into `<c:when test="${isOAuth2Configured}">`. I wouldn't expect the token to be populated when auth is not configured, so this made sense to me.

Otherwise, I think we can use a `c:if` over the how navbar, since these elements are not needed in any form inputs, so they can be not present rather than hidden with CSS.